### PR TITLE
Fix #853: Filter sidebar layout & mobile responsiveness improvements

### LINKED ISSUE #853: Redefine gallery filter options
With 235 filter options, the filters have become less helpful. Many of the filter options only have a single template that the filter applies to (i.e. of the 6 "topic" filters, 5 have less then 4 templates in their category). The search bar has updated to be able to search by template or author, making the author section redundant. 

Current filter options: 
- Collection (5)
- Language (9)
- Framework (22)
- Services (55)
- Database (8)
- Infrastructure as Code (2)
- Tools (17)
- Topic (6)
- Author (111)

### PR DESCRIPTION
**Pull request by hemarina**
Fixes #853. Partially addresses #667.

## Summary

Improves the gallery filter sidebar layout so that long category labels (notably **Infrastructure as Code**) fit on a single line, and makes the sidebar / "Viewing … Sort by" row behave correctly at smaller viewport widths.

### Filter sidebar (`ShowcaseLeftFilters`)
- Allow long category labels to wrap gracefully when needed (`white-space: normal; overflow-wrap: anywhere;` on checkbox list items).
- Add template counts next to **uncategorized** tags (AI Collection, Community Authored, Microsoft Authored, New, Popular) — same `Label (N)` pattern already used by `ShowcaseTagSelect` for other categories.
- Apply a minimum-count threshold (`MIN_TEMPLATE_COUNT = 3`) to **all tag categories** — Language, Framework, Services, Database, Infrastructure as Code, Tools, and Topic — so rarely-used tags are hidden by default. Currently-selected tags are always kept visible regardless of count.

### Sidebar width
- Bumped `.filter` `max-width` from `260px` → `270px` so the **Infrastructure as Code** header label fits on one line in the accordion.

### Mobile layout (≤ 996px)
- `.filterAndCard` switches to `flex-direction: column` with a `32px` gap between the filter bar and the "Viewing …" line (matches the existing spacing between the Viewing line and the cards).
- `.filter` becomes full width (`width: 100%; align-self: stretch; margin: 0; position: static;`) instead of being constrained to `max-width: 270px` and pinned with `align-self: flex-start`.
- `.card` `margin: 0` so the cards align flush with the filter bar (otherwise the desktop `margin: 0 0 0 20px` left-offsets the cards from the now-stacked filter).
- Category divider lines in the accordion are widened to full width and slightly thicker (`100% / 1px` vs `89.8% / 0.6px`) so they read correctly on the wider mobile sidebar.

### Viewing / Sort by row (≤ 547px)
- Added a `.viewingSortRow` class with `flex-wrap: wrap; row-gap: 8px;` so the **Sort by:** dropdown drops to a second line on narrow viewports instead of cramping the "Viewing 1‑20 of N templates" text. The 547px threshold is the width at which the two blocks begin to overlap with the current Combobox sizing.

## Files changed
- `website/src/components/gallery/ShowcaseLeftFilters/index.tsx`
- `website/src/components/gallery/ShowcaseLeftFilters/styles.module.css`
- `website/src/pages/ShowcaseCardPage.tsx`
- `website/src/pages/styles.module.css`

## Verification
- `npm run build` passes.
- Manually verified at viewport widths 1280px, 996px, 800px, 547px, 400px:
  - "Infrastructure as Code" stays on one line.
  - Uncategorized tags show counts.
  - Below 996px the filter bar is full width with proper spacing, full-width dividers, and cards aligned flush.
  - Below 547px "Sort by:" wraps to its own line.

### CODE REVIEW

**Comment by hemarina** [comment]
@haileyhuber8 Could you take a look at this PR? I'm open to other improvements for the filter bar.

**Review by jongio — COMMENTED** [review]
The filter threshold and mobile layout changes are straightforward. A few things:

1. The Tools/Infrastructure as Code categories are swapped (previously positions 5/6, now 6/5). If intentional, update the PR description; otherwise swap back.

2. ~150 lines of dead code remain from the Authors section removal (`ShowcaseAuthorFilterViewAll` component, `sortedAuthors`, author imports/props). Clean up in this PR or a fast follow-up.

3. Agree with the existing comment about `.card` left margin - it'll be offset 20px on mobile since `margin: 0` is only set on `.filter` in the <=996px query. Adding `.card { margin: 0; }` in the same media query should fix it.

4. The min-count threshold applies to all categories (Language, Framework, Services, Database, IaC, Tools, Topic) but the PR description only mentions Language. If the broader scope is intentional, update the description.

File: website/src/components/gallery/ShowcaseLeftFilters/index.tsx [hunk 1]
```diff
@@ -475,201 +476,223 @@ export default function ShowcaseLeftFilters({
...
+            <AccordionItem value="5">
+              <AccordionHeader
+                expandIconPosition="end"
+                style={{
+                  background:
+                    "linear-gradient(#D1D1D1 0 0) top /89.8% 0.6px no-repeat",
+                }}
+              >
+                <div style={{ fontSize: "16px", fontWeight: "500" }}>
+                  Tools
```
Comment (jongio) [inline website/src/components/gallery/ShowcaseLeftFilters/index.tsx:469]: This puts Tools at position 5 and IaC at position 6 - the previous order was reversed. Was the reorder intentional?

File: website/src/components/gallery/ShowcaseLeftFilters/index.tsx [hunk 2]
```diff
@@ -384,42 +384,43 @@ export default function ShowcaseLeftFilters({
     });
   }
   const sortedAuthors = Array.from(allAuthors).sort();
```
Comment (jongio) [inline website/src/components/gallery/ShowcaseLeftFilters/index.tsx]: `sortedAuthors` is computed but never used after the Authors accordion removal. `ShowcaseAuthorFilterViewAll` (lines 27-140), `ShowcaseAuthorSelect` import, `splitAuthors` import, and the author extraction loop (lines 373-385) are all dead code now.

File: website/src/components/gallery/ShowcaseLeftFilters/index.tsx (hunk 1 above)
Comment (hemarina) [reply website/src/components/gallery/ShowcaseLeftFilters/index.tsx:469]: Yes. Put IaC lower as it only has two tags, bicep vs terraforms. We only have 10 terraform templates.

File: website/src/components/gallery/ShowcaseLeftFilters/index.tsx [hunk 3]
```diff
@@ -384,42 +384,43 @@ export default function ShowcaseLeftFilters({
...
+  const MIN_TEMPLATE_COUNT = 3;
+  const applyMinCountThreshold = React.useCallback(
+    (tags: TagType[]): TagType[] =>
+      tags.filter(
+        (tag) => (tagCounts?.[tag] ?? 0) >= MIN_TEMPLATE_COUNT || selectedTags.includes(tag),
```
[the remark this answers is not in the document]
Comment (hemarina) [reply website/src/components/gallery/ShowcaseLeftFilters/index.tsx:238]: It is intended to exclude Language as some rare language templates deserve to be discovered.

**Review by jongio — COMMENTED** [review]
The "address feedback" commit handles most of the prior issues:

- `.card { margin: 0; }` on mobile: fixed
- AccordionHeader styling: the `categoryHeader` class approach is cleaner than the global selector override
- Tools/IaC order: understood, makes sense given IaC only has 2 tags
- Min-count threshold scope: understood, keeping Language unfiltered for discoverability

One item from my earlier review is still partially open: `ShowcaseAuthorFilterViewAll` (lines 26-140) and the `ShowcaseAuthorSelect` import (line 8) aren't referenced anywhere now that the Authors accordion is gone. The `sortedAuthors`/`allAuthors`/`splitAuthors` cleanup in this commit was good, but the component definition itself (~115 lines) is still there. Worth removing in this PR or a follow-up.

**Review by jongio — COMMENTED** [review]
Addresses my previous feedback. The dead code (ShowcaseAuthorFilterViewAll, ShowcaseAuthorSelect) is fully removed now. Looks good.

**Review by jongio — APPROVED** [review]
All feedback addressed. Approving.

**Review by wbreza — COMMENTED** [review]
## Code Review — PR #860

**TL;DR:** Improves gallery filter sidebar layout (wider, wrapping labels), adds template count thresholds to reduce low-signal filters, removes Authors accordion, and adds mobile responsiveness at two breakpoints. Net deletion of ~195 lines — solid cleanup.

> **Note:** Issues already flagged by Copilot and @jongio are excluded below. This review only surfaces **new findings**.

---

### 🔴 Must Fix

**1. Unused `UserState` import**
`website/src/components/gallery/ShowcaseLeftFilters/index.tsx`

`UserState` is imported from `"../ShowcaseTemplateSearch"` but never referenced in the file. Should be removed to avoid build warnings.

**2. Inconsistent `MIN_TEMPLATE_COUNT` threshold — `uncategoryTag` bypasses filtering**
`website/src/components/gallery/ShowcaseLeftFilters/index.tsx`

`applyMinCountThreshold` is applied to Language, Framework, Services, Database, IaC, Tools, and Topic tags — but `uncategoryTag` is filtered with a plain `.filter()` without the min-count threshold. This means uncategorized tags with <3 templates still appear, which is inconsistent with the PR's goal of reducing low-signal filter options per issue #853. If this exclusion is intentional (like Language), please add a comment explaining why. If not, apply `applyMinCountThreshold` to `uncategoryTag` as well.

---

### 🟡 Should Fix

**3. Missing conditional rendering for `uncategoryTag`**
`website/src/components/gallery/ShowcaseLeftFilters/index.tsx`

All other tag categories are now wrapped with `{categoryTag.length > 0 && (...)}` to hide empty sections — nice pattern. However, `uncategoryTag` lacks this guard. If the threshold fix from finding #2 is applied, `uncategoryTag` could become empty and would render an empty section.

**4. Several undocumented behavioral changes**
- **"View All" threshold** changed from `> 5` to `> 6` — the 6th tag now displays inline instead of behind the accordion. The change is correct (avoids a 1-item accordion), but not mentioned in the PR description.
- **Template counts added to all categories** including Infrastructure as Code — not mentioned.
- **`.categoryHeader` CSS class** added to all 7 AccordionHeaders with a mobile-specific gradient override — not documented.

Consider updating the PR description to capture these changes for future reference.

**5. `!important` on `.categoryHeader` suggests inline style should move to CSS**
`website/src/components/gallery/ShowcaseLeftFilters/styles.module.css`

The `!important` is needed because AccordionHeaders use inline `style={{ background: ... }}`. Moving the gradient styling entirely to the `.categoryHeader` CSS class would eliminate the need for `!important` and improve maintainability.

**6. Fragile `:global(.fui-Checkbox__label)` selector**
`website/src/components/gallery/ShowcaseLeftFilters/styles.module.css`

Targeting Fluent UI's internal class name directly couples to their implementation details. If Fluent UI renames this class in a future release, the word-wrap fix breaks silently. Consider adding a comment documenting this dependency.

---

### 🟢 Nitpick

**7. Hardcoded `gap: 32px` — consider using design system variable**
`website/src/pages/styles.module.css`

The repo defines spacing variables in `custom.css`. Using `gap: var(--space-xl)` would be more consistent with the design system.

**8. No tests for new `MIN_TEMPLATE_COUNT` filtering logic**
Not blocking for a CSS/layout PR, but worth a follow-up. Edge cases (0 templates, exactly 3, selected-but-below-threshold) are untested.

---

### ✅ Things Done Well
- Net deletion of ~195 lines — good cleanup
- Correct fix for "View All" threshold (5→6) to avoid 1-item accordions
- Baseline files properly updated
- CHANGELOG entry added
- Mobile responsive breakpoints are reasonable
- `overflow-wrap: anywhere` is the right fix for long label wrapping

---

**Overall:** Solid improvements to the filter UX. Two items need attention (unused import + threshold inconsistency), and a few description updates would help future maintainers. Nice work on the mobile responsiveness! 👍

**Comment by hemarina** [comment]
@copilot+claude-opus-4.6 Address the feedback: https://github.com/Azure/awesome-azd/pull/860#pullrequestreview-4239057046
